test/unit/cliUtil.test.js

```diff
@@ -0,0 +1,55 @@
+const cliUtil = require('../../src/util/cliUtil');
+const fs = require('fs');
+const path = require('path');
+
+const { SITE_JSON_DEFAULT } = require('./utils/data');
+
+jest.mock('fs');
+jest.mock('process');
+
+afterEach(() => {
+  fs.vol.reset();
+  jest.resetModules();
+});
+
+test('findRootFolder returns user specified root if site config is found there', () => {
+  const json = {
+    'userSpecifiedRoot/site.json': SITE_JSON_DEFAULT,
+  };
+  fs.vol.fromJSON(json, '');
+  const resolvedUserSpecificRoot = path.resolve('userSpecifiedRoot');
+  expect(cliUtil.findRootFolder('userSpecifiedRoot')).toBe(resolvedUserSpecificRoot);
+});
+
+test('findRootFolder throws error if site config is found in user specified root', () => {
+  const resolvedUserSpecificRoot = path.resolve('userSpecifiedRoot');
+  expect(
+    () => {
+      cliUtil.findRootFolder('userSpecifiedRoot');
+    })
+    .toThrow(`Config file not found at user specified root ${resolvedUserSpecificRoot}`);
+});
+
+test('findRootFolder without user root returns the closest parent directory containing site config', () => {
+  const json = {
+    './site.json': SITE_JSON_DEFAULT,
+    './nested/': {},
+  };
+  fs.vol.fromJSON(json, '/root');
+  process.cwd = jest.fn().mockReturnValue('/root/nested');
+  expect(cliUtil.findRootFolder()).toBe('/root');
```

Comment:
This test fails on Windows OS

![error2](https://user-images.githubusercontent.com/31084833/51424690-4ee00400-1c0c-11e9-8ba2-0981ff7c2275.PNG)